src/include/reconcile.i

```diff
@@ -43,50 +43,28 @@ __wt_rec_need_split(WT_RECONCILE *r, size_t len)
  *     Initialize an address timestamp triplet.
  */
 static inline void
-__wt_rec_addr_ts_init(WT_RECONCILE *r, wt_timestamp_t *start_durable_tsp,
-  wt_timestamp_t *oldest_start_tsp, uint64_t *oldest_start_txnp, wt_timestamp_t *stop_durable_tsp,
-  wt_timestamp_t *newest_stop_tsp, uint64_t *newest_stop_txnp, bool *preparep)
+__wt_rec_addr_ts_init(WT_RECONCILE *r, WT_TIME_AGGREGATE *ta)
 {
     /*
      * If the page is not fixed-length column-store, where we don't maintain timestamps at all, set
      * the oldest/newest timestamps to values at the end of their expected range so they're
      * corrected as we process key/value items. Otherwise, set the oldest/newest timestamps to
      * simple durability.
      */
-    *start_durable_tsp = WT_TS_NONE;
-    *oldest_start_tsp = WT_TS_MAX;
-    *oldest_start_txnp = WT_TXN_MAX;
-    *stop_durable_tsp = WT_TS_NONE;
-    *newest_stop_tsp = WT_TS_NONE;
-    *newest_stop_txnp = WT_TXN_NONE;
-    *preparep = false;
-    if (r->page->type == WT_PAGE_COL_FIX) {
-        *oldest_start_tsp = WT_TS_NONE;
-        *oldest_start_txnp = WT_TXN_NONE;
-        *newest_stop_tsp = WT_TS_MAX;
-        *newest_stop_txnp = WT_TXN_MAX;
-    }
+    if (r->page->type == WT_PAGE_COL_FIX)
+        __wt_time_aggregate_init(ta);
+    else
+        __wt_time_aggregate_init_max(ta);
 }
 
 /*
  * __wt_rec_addr_ts_update --
  *     Update the chunk's timestamp information.
  */
 static inline void
-__wt_rec_addr_ts_update(WT_RECONCILE *r, wt_timestamp_t start_durable_ts,
-  wt_timestamp_t oldest_start_ts, uint64_t oldest_start_txn, wt_timestamp_t stop_durable_ts,
-  wt_timestamp_t newest_stop_ts, uint64_t newest_stop_txn, bool prepare)
+__wt_rec_addr_ts_update(WT_RECONCILE *r, WT_TIME_WINDOW *tw)
 {
-    r->cur_ptr->newest_start_durable_ts =
-      WT_MAX(start_durable_ts, r->cur_ptr->newest_start_durable_ts);
-    r->cur_ptr->oldest_start_ts = WT_MIN(oldest_start_ts, r->cur_ptr->oldest_start_ts);
-    r->cur_ptr->oldest_start_txn = WT_MIN(oldest_start_txn, r->cur_ptr->oldest_start_txn);
-    r->cur_ptr->newest_stop_durable_ts =
-      WT_MAX(stop_durable_ts, r->cur_ptr->newest_stop_durable_ts);
-    r->cur_ptr->newest_stop_ts = WT_MAX(newest_stop_ts, r->cur_ptr->newest_stop_ts);
-    r->cur_ptr->newest_stop_txn = WT_MAX(newest_stop_txn, r->cur_ptr->newest_stop_txn);
-    if (prepare)
-        r->cur_ptr->prepare = true;
+    __wt_time_aggregate_update(r->cur_ptr->ta, tw);
```

Comment:
This could be a direct call it looks like.

Reply:
Fixed, pushed in 031541d8a.